homeassistant/components/motionblinds_ble/cover.py

```diff
@@ -0,0 +1,232 @@
+
+    @callback
+    def async_update_position(
+        self,
+        position: int | None,
+        tilt: int | None,
+    ) -> None:
+        """Update the position of the motor."""
+        if position is None:
+            self._attr_current_cover_position = None
+            self._attr_is_closed = None
+        else:
+            self._attr_current_cover_position = 100 - position
+            self._attr_is_closed = self._attr_current_cover_position == 0
+        if tilt is None:
+            self._attr_current_cover_tilt_position = None
+        else:
+            self._attr_current_cover_tilt_position = 100 - round(100 * tilt / 180)
+        self.async_write_ha_state()
+
+
+class PositionBlind(MotionblindsBLECoverEntity):
+    """Representation of a blind with position capability."""
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN
+        | CoverEntityFeature.CLOSE
+        | CoverEntityFeature.STOP
+        | CoverEntityFeature.SET_POSITION
+    )
+
+    async def async_open_cover(self, **kwargs: Any) -> None:
+        """Open the blind."""
+        _LOGGER.debug("(%s) Opening", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.open()
+
+    async def async_close_cover(self, **kwargs: Any) -> None:
+        """Close the blind."""
+        _LOGGER.debug("(%s) Closing", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.close()
+
+    async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Move the blind to a specific position."""
+        if kwargs.get(ATTR_POSITION) is None:
+            return
+        new_position: int = 100 - int(kwargs[ATTR_POSITION])
+
+        _LOGGER.debug(
+            "(%s) Setting position to %i",
+            self.config_entry.data[CONF_MAC_CODE],
+            new_position,
+        )
+        await self._device.position(new_position)
+
+
+class TiltBlind(MotionblindsBLECoverEntity):
+    """Representation of a blind with tilt capability."""
+
+    _attr_supported_features = (
+        CoverEntityFeature.OPEN_TILT
+        | CoverEntityFeature.CLOSE_TILT
+        | CoverEntityFeature.STOP_TILT
+        | CoverEntityFeature.SET_TILT_POSITION
+    )
+
+    async def async_open_cover_tilt(self, **kwargs: Any) -> None:
+        """Tilt the blind open."""
+        _LOGGER.debug("(%s) Tilt opening", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.open_tilt()
+
+    async def async_close_cover_tilt(self, **kwargs: Any) -> None:
+        """Tilt the blind closed."""
+        _LOGGER.debug("(%s) Tilt closing", self.config_entry.data[CONF_MAC_CODE])
+        await self._device.close_tilt()
+
+    async def async_stop_cover_tilt(self, **kwargs: Any) -> None:
+        """Stop tilting the blind."""
+        await self.async_stop_cover(**kwargs)
+
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
+        """Tilt the blind to a specific position."""
+        if kwargs.get(ATTR_TILT_POSITION) is None:
```

Comment:
Not sure this is reachable, please check the schema in the base entity to see if it's validated already

Reply:
Other integrations don't seem to check it so I removed it